os/mm/mm_heap/mm_manage_allocfail.c

```diff
@@ -66,8 +60,27 @@
  *
  ************************************************************************/
 
+#if defined(CONFIG_APP_BINARY_SEPARATION) && !defined(__KERNEL__)
+void mm_ioctl_alloc_fail(size_t size)
+{
+	int mmfd = open(MMINFO_DRVPATH, O_RDWR);
+	if (mmfd < 0) {
+		mdbg("Fail to open %s, errno %d\n", MMINFO_DRVPATH, get_errno());
+	} else {
+		int res = ioctl(mmfd, MMINFOIOC_MNG_ALLOCFAIL, (int)size);
+		if (res == ERROR) {
+			mdbg("Fail to call mm_manage_allocfail, errno %d\n", get_errno());
+		}
+		close(mmfd);
+	}
+}
+
+#else
+
 void mm_manage_alloc_fail(struct mm_heap_s *heap, int startidx, int endidx, size_t size, int heap_type)
 {
+	(void)irqsave();
```

Comment:
There is no irqrestore which is matched with this line. When we disable CONFIG_MM_ASSERT_ON_FAIL, this function won't cause device reset, it's just mm failure. So I think irqrestore is necessary. Could you check?

Reply:
Added irqrestore at the the end of function.